single test fail assertion `1

ubuntu 18.04 gcc version 7.3.0 single core 4gb ram
commit 1f96584c940559507bfb49fb45e20f5e59183804

testAll: /home/phaverkamp/loki-network/llarp/queue_manager.cpp:338: llarp::thread::QueueReturn llarp::thread::QueueManager::reservePopIndex(uint32_t&, uint32_t&): Assertion `1 == (circularDifference(currGen, elemGen, m_maxGeneration) + 1)' failed.
testAll: /home/phaverkamp/loki-network/llarp/queue_manager.cpp:338: llarp::thread::QueueReturn llarp::thread::QueueManager::reservePopIndex(uint32_t&, uint32_t&): Assertion `1 == (circularDifference(currGen, elemGen, m_maxGeneration) + 1)' failed.
Makefile:149: recipe for target 'test' failed
make: *** [test] Aborted (core dumped)

--- a/llarp/queue_manager.cpp
+++ b/llarp/queue_manager.cpp
@@ -335,7 +335,7 @@ namespace llarp
           // Queue is full.
           assert(state == ElementState::Reading);
           assert(
-              1 == (circularDifference(currGen, elemGen, m_maxGeneration) + 1));
+              1 == (circularDifference(currGen, elemGen, m_maxGeneration + 1)));
 
           return QueueReturn::QueueEmpty;
         }

